packages/utils/src/types.tsx

```diff
@@ -82,8 +82,15 @@ export type ComponentWithForwardedRef<
   ComponentProps
 > = React.ForwardRefExoticComponent<
   ComponentProps &
-    React.HTMLProps<React.ElementType<ElementType>> &
-    React.ComponentPropsWithRef<ElementType>
+    // Need to omit data from html props to allow a custom prop type in components.
+    // Otherwise, it always falls back to the html element attribute data: string for
+    // attaching a url of an element. However, I have never seen this used before.
+    // https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes
+    Omit<
```

Comment:
I really don't love this but the only alternative for allowing a custom "data" prop in components would be to either create a new `ComponentWithForwardedRef` type specifically for the `ComboboxOption` (right now, the component seems to be the only one using this type util anyway), or to find a different prop name than `data`.

The `data` prop of HTML elements looks like a legacy HTML prop to me. It isn't even documented on mdn (only on [w3c](https://www.w3schools.com/tags/att_object_data.asp)) and I don't think people would ever use it for developing React applications. How should we go about this?

Reply:
I'd personally be in preference of preserving native attributes and naming props something that doesn't conflict. This also keeps us in line with the philosophy of [pretending we're HTML][1].

[1]: https://gist.github.com/ryanflorence/e5c794e6093d16a69fa88d2112a292f7#random-implementation-checklist

Reply:
Sure. Any ideas for a good name?

Only two I can think off at the moment that are not already taken by HTML are:

- `selectData`
- `context`

Reply:
Went for `selectData`